backend/chalice/api_server/run_local_server.py

```diff
@@ -10,9 +10,35 @@
 import os
 import sys
 import time
+import signal
 
 from chalice.deploy.validate import validate_routes
 from chalice.cli import CLIFactory, reloader
+from chalice.local import LocalDevServer
+
+from six.moves.socketserver import ForkingMixIn
+from six.moves.BaseHTTPServer import HTTPServer
+
+
+class ForkedHTTPServer(ForkingMixIn, HTTPServer):
+    """Forking mixin to better support browsers.
+
+    When a browser sends a GET request to Chalice it keeps the connection open
+    for reuse. In the single threaded model this causes Chalice local to become
+    unresponsive to all clients other than that browser socket. Even sending a
+    header requesting that the client close the connection is not good enough,
+    the browswer will simply open another one and sit on it.
+    """
+    allow_reuse_address = True
+    timeout = 2
+
+    # Make sure to reap child processes.
+    def shutdown(self):
+        if self.active_children is None:
+            return
+        for child_pid in self.active_children:
+            os.kill(child_pid, signal.SIGTERM)
+        self.server_close()
```

Comment:
The data portal application is not thread safe, and the default local server runs in threaded mode. This was causing intermittent 500 errors as db connections were getting cleared mid-request when multiple requests were being processed in parallel.

Switching to a forking webserver reduces unexpected failures.